src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -119,8 +121,35 @@ void TransposedConvolution<
   inputTemp = arma::cube(const_cast<arma::Mat<eT>&&>(input).memptr(),
       inputWidth, inputHeight, inSize * batchSize, false, false);
 
-  outputWidth = TransposedConvOutSize(inputWidth, kW, dW, padW);
-  outputHeight = TransposedConvOutSize(inputHeight, kH, dH, padH);
+  aW = (outputWidth + kW - 2 * padW - 2) % dW;
+  aH = (outputHeight + kH - 2 * padH - 2) % dH;
+
+  // Check if the output height and width are possible given the other
+  // parameters of the layer.
+  if (outputWidth != dW * (inputWidth - 1) + aW + 2 * padW + 2 - kW ||
+      outputHeight != dH * (inputHeight - 1) + aW + 2 * padH + 2 - kH)
+  {
+    Log::Fatal << "The output width / output height is not possible given "
+        << "the other parameters of the layer." << std::endl;
+  }
+
+  if (dW > 1 || dH > 1)
+  {
+    InsertZeros(inputTemp, dW, dH, inputExpandedTemp);
+
+    if (padW != 0 || padH != 0 || aW != 0 || aH != 0)
+      Pad(inputExpandedTemp, padW, padH, aW, aH, inputPaddedTemp);
+    else
+    {
+      inputPaddedTemp = arma::Cube<eT>(inputExpandedTemp.memptr(),
```

Comment:
I think we can optimize this code. But this will change after using the `padding` layer.